great_expectations/expectations/expectation.py

```diff
@@ -694,6 +698,206 @@ def build_configuration(cls, *args, **kwargs):
+                sub_result_list.append(self._get_rendered_result_as_string(sub_result))
+
+            return "\n".join(sub_result_list)
+
+        elif type(rendered_result) == RenderedStringTemplateContent:
+            return rendered_result.__str__()
+
+        else:
+            print(type(rendered_result))
+
+    def _get_renderer_dict(
+        self,
+        expectation_name: str,
+        expectation_config: ExpectationConfiguration,
+        validation_result: ExpectationValidationResult,
+        standard_renderers=[
+            "renderer.answer",
+            "renderer.diagnostic.unexpected_statement",
+            "renderer.diagnostic.observed_value",
+            "renderer.diagnostic.status_icon",
+            "renderer.diagnostic.unexpected_table",
+            "renderer.prescriptive",
+            "renderer.question",
+        ],
+    ) -> Dict[str, str]:
+        supported_renderers = self._get_supported_renderers(expectation_name)
+
+        standard_renderer_dict = {}
+
+        for renderer_name in standard_renderers:
+            if renderer_name in supported_renderers:
+                _, renderer = _registered_renderers[expectation_name][renderer_name]
+
+                rendered_result = renderer(
+                    configuration=expectation_config,
+                    result=validation_result,
+                )
+                standard_renderer_dict[
+                    renderer_name
+                ] = self._get_rendered_result_as_string(rendered_result)
+
+            else:
+                standard_renderer_dict[renderer_name] = None
+
+        return {
+            "standard": standard_renderer_dict,
+            "custom": [],
+        }
+
+    def _get_execution_engine_dict(
+        self,
+        upstream_metrics,
+    ) -> Dict:
+        expectation_engines = {}
+        for provider in [
+            "PandasExecutionEngine",
+            "SqlAlchemyExecutionEngine",
+            "SparkDFExecutionEngine",
+        ]:
+            all_true = True
+            for metric in upstream_metrics:
+                if not provider in _registered_metrics[metric]["providers"]:
+                    all_true = False
+                    break
+
+            expectation_engines[provider] = all_true
+
+        return expectation_engines
+
+    def _get_upstream_metrics(self, expectation_config) -> List[str]:
+        # NOTE: Abe 20210102: Strictly speaking, identifying upstream metrics shouldn't need to rely on an expectation config.
+        # There's probably some part of get_validation_dependencies that can be factored out to remove the dependency.
+        validation_dependencies = self.get_validation_dependencies(
+            configuration=expectation_config
+        )
+
+        return list(validation_dependencies["metrics"].keys())
+
+    def _get_library_metadata(self):
+        if hasattr(self, "library_metadata"):
+            return self.library_metadata
+        else:
+            return {}
+
```

Comment:
This is great -- and will not be in conflict with the changes to `test_yaml_config` and the introduction of additional `self_check` methods added as part of the Checkpoints pod.